components/AddToPlaylist.vue

```diff
@@ -0,0 +1,73 @@
+<script lang="ts" setup>
+import { ResponseError } from "@bcc-code/bmm-sdk-fetch";
+
+const { t } = useI18n();
+
+const props = defineProps<{
+  listSelected: (selectedPlaylistId: number) => Promise<void>;
+}>();
+const emit = defineEmits<{
+  close: [];
+}>();
+
+const { data: playlists } = usePrivatePlaylists();
+
+const selectList = async (playlistId: number, playlistName: string) => {
+  try {
+    const promise = props.listSelected(playlistId);
+    emit("close");
```

Comment:
Isn't it better to close the dialog _after_ the promise has resolved?

Reply:
it currently works as an optimistic UI.